### PR TITLE
refactor: pass operation objects to adapters

### PR DESCRIPTION
**Pull request by justinsb**
This should allow for richer behaviours, such as intermediate updates of status.

### CODE REVIEW

**Comment by justinsb** [comment]
I'm experimenting with using this in the DataflowFlexTemplateJob controller in #2580 , if you want to peek at the last commit...

**Comment by jasonvigil** [comment]
/approve

Looks like there are some failing tests though

File: pkg/controller/direct/directbase/operations.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	object *unstructured.Unstructured
+}
+
+func NewUpdateOperation(object *unstructured.Unstructured) *UpdateOperation {
+	return &UpdateOperation{
+		object: object,
+	}
+}
+
+func (o *UpdateOperation) UnstructuredObject() *unstructured.Unstructured {
```
Comment (yuwenma) [inline pkg/controller/direct/directbase/operations.go:15]: My first read from the fn name  is that I thought you mean the `unstructured.Unstructured` `Object`. https://github.com/kubernetes/apimachinery/blob/v0.31.0/pkg/apis/meta/v1/unstructured/unstructured.go#L45C2-L45C8 
Can we rename this something like `Unstructured()` or `GetUnstructured()`?

**Review by yuwenma — COMMENTED** [review]
/lgtm
/approve

File: pkg/controller/direct/directbase/operations.go (hunk 1 above)
Comment (justinsb) [reply pkg/controller/direct/directbase/operations.go]: Good call, done!

**Review by jasonvigil — APPROVED** [review]

**Pull request merged**